framework/src/engine/network/endpoint.ts

```diff
@@ -33,10 +33,14 @@ export class Endpoint {
 			const parsedPeer = {
 				...peer,
 			};
+
+			parsedPeer.chainID = parsedPeer.chainID?.toString('hex') as unknown as Buffer;
```

Comment:
i don't really understand this conversion. It is saying `toString` but type is `Buffer` 🤔 
also, you could do
```
const parsedPeer = {
	...peer,
	chainID: peers.chainID?.toString('hex') ?? '',
};
```

Reply:
This works too 🙂